spec/requests/cats_spec.rb

```diff
@@ -49,39 +49,217 @@
-       cat: {
-        name: 'Mr.Johnson',
-        age: 7,
-        enjoys: 'Furrrrociously hunting bugs.',
-        image: 'https://c0.wallpaperflare.com/preview/135/654/456/italy-epic-cat-cats.jpg'
-  }
-  }
+  updated_cat_params = {
+    cat: {
+    name: 'Mr.Johnson',
+    age: 7,
+    enjoys: 'Furrrrociously hunting bugs.',
+    image: 'https://c0.wallpaperflare.com/preview/135/654/456/italy-epic-cat-cats.jpg'
+}
+}
       
     cat = Cat.first
     patch "/cats/#{cat.id}", params: updated_cat_params
         
-
-        expect(response).to have_http_status(200)
-        updated_cat = Cat.find(cat.id)
-        expect(updated_cat.age).to eq 7
-   end
+    expect(response).to have_http_status(200)
+    updated_cat = Cat.find(cat.id)
+    expect(updated_cat.age).to eq 7
     end
+  end
 
-    describe "DELETE / destroy" do
-      it "deletes a cat" do
+  describe "DELETE / destroy" do
+   it "deletes a cat" do
         Cat.create(
-        {
+    {
+    name: 'Mr.Johnson',
+    age: 5,
+    enjoys: 'Furrrrociously hunting bugs.',
+    image: 'https://c0.wallpaperflare.com/preview/135/654/456/italy-epic-cat-cats.jpg'
+}
+        )
+    cat = Cat.first
+    delete "/cats/#{cat.id}"
+    
+    expect(response).to have_http_status(200)
+      end
+      end
+           
+   describe "can not create cat without valid attributes" do
+    it "does not creat cat without name" do
+      cat_params = {
+    cat: {
+    age: 5,
+    enjoys: 'Furrrrociously hunting bugs.',
+    image: 'https://c0.wallpaperflare.com/preview/135/654/456/italy-epic-cat-cats.jpg'
+      }
+    }
+    post "/cats", params: cat_params
+
+    expect(response.status).to eq 422
+    cat = JSON. parse(response.body)
+    expect(cat['name']).to include "can't be blank"
+    end
+
+
+    it "does not create cat without age" do
+    cat_params = {
+    cat: {
+    name: 'Mr.Johnson',
+    enjoys: 'Furrrrociously hunting bugs.',
+    image: 'https://c0.wallpaperflare.com/preview/135/654/456/italy-epic-cat-cats.jpg'
+    }
+        }
+  post "/cats", params: cat_params
+
+  expect(response.status).to eq 422
+  cat = JSON. parse(response.body)
+  expect(cat['age']).to include "can't be blank"
+    end
+
+    it "does not create cat without enjoys" do
+    cat_params = {
+    cat: {
+    age: 5,
+    name: 'Mr.Johnson',
+    image: 'https://c0.wallpaperflare.com/preview/135/654/456/italy-epic-cat-cats.jpg'
+        }
+         }
+   post "/cats", params: cat_params
+
+   expect(response.status).to eq 422
+   cat = JSON. parse(response.body)
+   expect(cat['enjoys']).to include "can't be blank"
+     end
+
+   it "does not create cat without image" do
+    cat_params = {
+      cat: {
+      age: 5,
       name: 'Mr.Johnson',
+      enjoys: 'Furrrrociously hunting bugs.',
+      }
+    }
+    post "/cats", params: cat_params
+
+    expect(response.status).to eq 422
+    cat = JSON. parse(response.body)
+    expect(cat['image']).to include "can't be blank"
+      end
+     end
```

Comment:
The ends for the blocks should not be further over than the code it encompasses.